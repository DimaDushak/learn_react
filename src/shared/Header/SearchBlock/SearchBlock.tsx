import React from 'react';
import styles from './searchblock.css';
import UserBlock from '../../UserBlock';

export default function SearchBlock(): JSX.Element {
    return (
        <div className={styles.searchBlock}>
            <UserBlock className={styles.userBlock} />
        </div>
    );
}
