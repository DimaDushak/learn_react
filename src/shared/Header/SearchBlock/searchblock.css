.searchBlock {
    display: flex;
}

.userBlock:hover img {
    box-shadow: 0 0 0 1px var(--orange);
}

.userBlock:hover span {
    color: var(--orange);
}

.userBlock img {
    height: 30px;
    padding: 2px;
    transition: box-shadow 0.3s;
    width: 30px;
}

.userBlock span {
    display: none;
}

@media (min-width: 1024px) {
    .searchBlock {
        flex-direction: row-reverse;
    }

    .userBlock img {
        height: 50px;
        margin-right: 10px;
        width: 50px;
    }

    .userBlock span {
        display: block;
        font-size: 1rem;
        line-height: 1.2rem;
        transition: color 0.3s;
    }
}
