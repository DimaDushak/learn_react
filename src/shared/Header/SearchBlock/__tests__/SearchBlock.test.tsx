import { render } from '@testing-library/react';
import React from 'react';
import SearchBlock from '../SearchBlock';

test('should render SearchBlock (snapshot)', () => {
    const { container } = render(<SearchBlock />);

    expect(container.firstChild).toMatchSnapshot();
});
