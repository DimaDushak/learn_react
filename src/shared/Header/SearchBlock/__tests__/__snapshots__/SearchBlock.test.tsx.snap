// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBlock (snapshot) 1`] = `
<div>
  search
</div>
`;
